### PR TITLE
feat(search-doc): search over multiple fields, readme update

### PR DESCRIPTION
**Pull request by AnguIar**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,39 +1,22 @@
-# Map Colonies typescript service template
+# CommonStorage
 
-This is a basic template for building new map colonies services in typescript.
+CommonStorage is a Node.js & Elasticsearch service for storing and handling static data sets, and is best for when you want to do fast full-text searches over your application data.
```
Comment (asafMasa) [inline readme.md:3]: improve the syntax

File: src/controllers/indexes.ts [hunk 2]
```diff
@@ -78,47 +79,71 @@ export class IndexesController {
...
       const { name }: any = req.params;
-      const match: any = req.query;
 
-      // If no request query, return all documents
-      const query : any = Object.keys(match).length ? { match } : { match_all: {} };
+      // We get a key:value seearchobject that contains the request parameters
+      // and we parser it to elastic typed search object
+      // https://stackoverflow.com/questions/39507005/query-with-match-by-multiple-fields
+
+      const queryToElastic = (query: any) => {
```
Comment (asafMasa) [inline src/controllers/indexes.ts:113]: refactor to private method

File: src/routers/indexes.ts [hunk 3]
```diff
@@ -12,7 +12,8 @@ indexesRouter.post('/', validate, controller.create.bind(controller));
 indexesRouter.delete('/:name', validate, controller.delete.bind(controller));
 
 indexesRouter.post('/:name/document', validate, controller.addDocument.bind(controller));
-indexesRouter.delete('/:name/document/:id', validate, controller.delete.bind(controller));
-indexesRouter.get('/:name/document', validate, controller.searchDocument.bind(controller));
+indexesRouter.delete('/:name/document/:documentId', validate, controller.deleteDocument.bind(controller));
+// indexesRouter.get('/:name/document', validate, controller.searchDocument.bind(controller));
```
Comment (asafMasa) [inline src/routers/indexes.ts:16]: add todo: xxxxxx
open issue with open-api validator

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: readme.md (hunk 1 above)
Comment (AnguIar) [reply readme.md:3]: Not for current issue

File: src/routers/indexes.ts (hunk 3 above)
Comment (AnguIar) [reply src/routers/indexes.ts]: Issue opened at https://github.com/PayU/openapi-validator-middleware/issues/140

File: src/controllers/indexes.ts [hunk 4]
```diff
@@ -106,23 +106,7 @@ export class IndexesController {
...
-          });
-          return { bool: { must: match } };
-        } else {
-          // If no query parameter is supplied, return all documents
-          return { match_all: {} };
-        }
-      };
-
-      const queryFixed: object = queryToElastic(req.query);
+      const queryFixed: object = this.queryToElastic(req.query);
```
Comment (asafMasa) [inline src/controllers/indexes.ts:109]: const elasticQuery: object = this.buildElasticQuery(req.query);

**Review by asafMasa — APPROVED** [review]
💯

**Review by asafMasa — APPROVED** [review]

**Pull request merged**